supertokens_python/recipe/totp/interfaces.py

```diff
@@ -0,0 +1,268 @@
+        DeviceAlreadyExistsError,
+        UnknownDeviceError,
+    ]:
+        pass
+
+    @abstractmethod
+    async def remove_device(
+        self, user_id: str, device_name: str
+    ) -> Union[RemoveTotpDeviceOkResult, TotpNotEnabledError]:
+        pass
+
+    @abstractmethod
+    async def list_devices(
+        self, user_id: str
+    ) -> Union[ListTotpDevicesOkResult, TotpNotEnabledError]:
+        pass
+
+
+class DeviceAlreadyExistsErrorResponse(APIResponse):
+    status = "DEVICE_ALREADY_EXISTS_ERROR"
+
+    def to_json(self):
+        return {"status": self.status}
+
+
+class InvalidTotpErrorReponse(APIResponse):
+    status = "INVALID_TOTP_ERROR"
+
+    def to_json(self):
+        return {"status": self.status}
+
+
+class TotpNotEnabledErrorResponse(APIResponse):
+    status = "TOTP_NOT_ENABLED_ERROR"
+
+    def to_json(self):
+        return {"status": self.status}
+
+
+class LimitReachedErrorResponse(APIResponse):
+    status = "LIMIT_REACHED_ERROR"
+
+    def to_json(self):
+        return {"status": self.status}
+
+
+class UnknownDeviceErrorResponse(APIResponse):
+    status = "UNKNOWN_DEVICE_ERROR"
+
+    def to_json(self):
+        return {"status": self.status}
+
+
+class CreateDevicePostOkResponse(APIResponse):
+    status = "OK"
+
+    def __init__(self, user_identifier: str, issuer_name: str, secret: str):
+        self.qr = f"otpauth://totp/{quote(issuer_name)}:{quote(user_identifier)}?secret={secret}&issuer={quote(issuer_name)}"
+
+    def to_json(self):
+        return {"status": self.status, "qr": self.qr}
+
+
+class VerifyCodePostOkResponse(APIResponse):
+    status = "OK"
+
+    def to_json(self):
+        return {"status": self.status}
+
+
+class VerifyDeviceOkResponse(APIResponse):
+    status = "OK"
+
+    def __init__(self, was_already_verified: bool):
+        self.was_already_verified = was_already_verified
+
+    def to_json(self):
+        return {"status": self.status, "wasAlreadyVerified": self.was_already_verified}
+
+
+class UpdateDevicePutOkResponse(APIResponse):
+    status = "OK"
+
+    def to_json(self):
+        return {"status": self.status}
+
+
+class RemoveDevicePostOkResponse(APIResponse):
+    status = "OK"
+
+    def __init__(self, did_exist: bool):
+        self.did_exist = did_exist
+
+    def to_json(self):
+        return {"status": self.status, "didExist": self.did_exist}
+
+
+class ListDevicesGetOkResponse(APIResponse):
+    status = "OK"
+
+    def __init__(self, devices: List[Dict[str, Any]]):
+        self.devices = devices
+
+    def to_json(self):
+        return {"status": self.status, "devices": self.devices}
+
+
+class APIInterface:
+    def __init__(self) -> None:
+        self.disable_create_device_post = False
+        self.disable_verify_code_post = False
+        self.disable_verify_device_post = False
+        self.disable_update_device_put = False
+        self.disable_remove_device_post = False
+        self.disable_list_devices_get = False
+
```

Comment:
all these APIs should not take a user_id. They should take a session container instead.